### PR TITLE
Added athorization for teams

### PR DESCRIPTION
**Pull request by Dmitry-Erokhin**
# One-line summary

in December'21 Nakadi introduced authorization on the __team__ level in addition to users and services. This change supports this feature from Clin side.

## Description
New section `teams` added to `auth` root element in event type, subscription, and sql-query yamls:

```yaml
  auth:
    teams:
      admins:
        - your_team
      readers:
        - sibling_team
      writers:
```

This section is optional to make existing definitions backward-compatible.

### Conraband
- updated `SECURITY.md` with relevant contact information (#75)
- bumped dependencies

## Types of Changes
- New feature (non-breaking change which adds functionality)
- Documentation / non-code

### CODE REVIEW

**Comment by dlippok** [comment]
Would you like to add some tests to this PR? At least updating `test_models.py` would make sense IMHO.

**Review by dlippok — COMMENTED** [review]
Would you like to add some tests to this PR? At least updating test_models.py would make sense IMHO.

**Comment by Dmitry-Erokhin** [comment]
@dlippok Done, sorry for taking that long for these 5 lines (=

File: tests/test_models.py [hunk 1]
```diff
@@ -7,6 +7,16 @@
 
 yml = """
 auth:
+  teams:
+    admins:
+      - your_team
+    readers:
+      - sibling_team
+      - - other_team_1
+        - other_team_2
```
Comment (dlippok) [inline tests/test_models.py:16]: It seems to work but is it intended? I am not sure if this "hierarchy" makes sense here.

Comment (Dmitry-Erokhin) [reply tests/test_models.py:16]: This is not hierarchy per se, this is workaround for the cases like:
```yaml
READER_TEAMS:
  - sibling_team
  - @@@./teams/teams_set_2.yaml #in file - [other_team_1, other team_2]
```
then the result would be nested:
```
- sibling team
- - other_team_1
- - other_team_2
```
however effectively we want tit flatterned.

**Comment by dlippok** [comment]
:+1:

**Comment by Dmitry-Erokhin** [comment]
👍

**Pull request merged**